File: data/EXP01/golang_no_system_var/2410/2410.output_3.go
func matchPlayersAndTrainers(players []int, trainers []int) int {
    sort.Ints(players)
    sort.Ints(trainers)

    playerPtr := 0
    trainerPtr := 0
    count := 0

    for playerPtr < len(players) && trainerPtr < len(trainers) {
        if players[playerPtr] <= trainers[trainerPtr] {
            count++
            playerPtr++
            trainerPtr++
        } else {
            trainerPtr++
        }
    }

    return count
}

File: data/EXP01/python/2410/2410.output_3.py
# @lc app=leetcode id=2410 lang=python3
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort()
        trainers.sort()
        
        count = 0
        for player in players:
            for i in range(len(trainers)):
                if trainers[i] >= player:
                    count += 1
                    trainers[i] = -1  # Mark the trainer as used
                    break
        
        return count